include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -139,43 +150,26 @@ typename MatType::elem_type NSGA2::Optimize(
     BinaryTournamentSelection(population, lowerBound, upperBound);
 
     // Evaluate the objectives for the new population.
-    calculatedObjectives.resize(population.size());
+    populationFval.resize(population.size());
     for (size_t i = 0; i < population.size(); i++)
-      calculatedObjectives[i] = arma::Col<ElemType>(numObjectives, arma::fill::zeros);
-    EvaluateObjectives(population, objectives, calculatedObjectives);
+      populationFval[i] = arma::Col<ElemType>(numObjectives, arma::fill::zeros);
+    EvaluateObjectives(population, objectives, populationFval);
 
     // Perform fast non dominated sort on P_t ∪ G_t.
     ranks.resize(population.size());
-    FastNonDominatedSort<MatType>(fronts, ranks, calculatedObjectives);
+    FastNonDominatedSort<MatType>(fronts, ranks, populationFval);
 
     // Perform crowding distance assignment.
     crowdingDistance.resize(population.size());
-
+    std::fill(crowdingDistance.begin(), crowdingDistance.end(), 0.f);
     for (size_t fNum = 0; fNum < fronts.size(); fNum++)
     {
-      CrowdingDistanceAssignment(fronts[fNum], crowdingDistance);
-    }
-
-    // Sort based on crowding distance.
-    std::sort(population.begin(), population.end(),
-              [this, ranks, crowdingDistance, population](MatType candidateP,
-                                                          MatType candidateQ)
-              {
-                size_t idxP, idxQ;
-                for (size_t i = 0; i < population.size(); i++)
-                {
-                  if (arma::approx_equal(population[i], candidateP, "absdiff", epsilon))
-                    idxP = i;
-
-                  if (arma::approx_equal(population[i], candidateQ, "absdiff", epsilon))
-                    idxQ = i;
-                }
-
-                return CrowdingOperator(idxP, idxQ, ranks, crowdingDistance);
-              }
-    );
+		  CrowdingDistanceAssignment<MatType>(
+			  fronts[fNum], populationFval, crowdingDistance);
```

Comment:
Looks like the indentation is wrong here, it should be two spaces, and the next line should tabbed twice (4 spaces).